pkg/kubelet/config/config.go

```diff
@@ -293,21 +295,24 @@ func (s *podStorage) seenSources(sources ...string) bool {
 func filterInvalidPods(pods []api.BoundPod, source string) (filtered []*api.BoundPod) {
 	names := util.StringSet{}
 	for i := range pods {
-		var errors []error
-		name := podUniqueName(&pods[i])
+		pod := &pods[i]
+		var errlist []error
+		name := podUniqueName(pod)
 		if names.Has(name) {
-			errors = append(errors, apierrs.NewFieldDuplicate("name", pods[i].Name))
+			errlist = append(errlist, apierrs.NewFieldDuplicate("name", pod.Name))
 		} else {
 			names.Insert(name)
 		}
-		if errs := validation.ValidateBoundPod(&pods[i]); len(errs) != 0 {
-			errors = append(errors, errs...)
+		if errs := validation.ValidateBoundPod(pod); len(errs) != 0 {
+			errlist = append(errlist, errs...)
 		}
-		if len(errors) > 0 {
-			glog.Warningf("Pod %d (%s) from %s failed validation, ignoring: %v", i+1, pods[i].Name, source, errors)
+		if len(errlist) > 0 {
+			glog.Warningf("Pod %d (%s) from %s failed validation, ignoring: %v", i+1, pod.Name, source, errlist)
+			//FIXME: is it kosher to log an event about a pod that might be invalid?
```

Comment:
I added a "best effort name" function, so we never log an empty name

On Fri, Jan 9, 2015 at 6:32 PM, Tim Hockin <thockin@google.com> wrote:

> Are we concerned that name or uid might be empty?
> On Jan 9, 2015 5:47 PM, "Daniel Smith" <notifications@github.com> wrote:
>
>> In pkg/kubelet/config/config.go
>> <https://github.com/GoogleCloudPlatform/kubernetes/pull/3382#discussion-diff-22754825>
>> :
>>
>> >  		}
>> > -		if len(errors) > 0 {
>> > -			glog.Warningf("Pod %d (%s) from %s failed validation, ignoring: %v", i+1, pods[i].Name, source, errors)
>> > +		if len(errlist) > 0 {
>> > +			glog.Warningf("Pod %d (%s) from %s failed validation, ignoring: %v", i+1, pod.Name, source, errlist)
>> > +			//FIXME: is it kosher to log an event about a pod that might be invalid?
>>
>> Yes, this is a great idea. Otherwise users will get no feedback when
>> kubelet ignores their pods.
>>
>> --
>> Reply to this email directly or view it on GitHub
>> <https://github.com/GoogleCloudPlatform/kubernetes/pull/3382/files#r22754825>
>> .
>>
>